ts/parsers/elementparser.ts

```diff
@@ -7,49 +7,134 @@ import * as isEmpty from "lodash.isempty";
 /**
  * Entry point for all Parsers
  */
+const SUPPORTED_PLACEHOLDERS = ["body", "ctrTitle", "pic", "subTitle", "tbl", "title"];
 class PowerpointElementParser {
     private element;
 
-    public getProcessedElement(rawElement, slideRelationships): PowerpointElement {
+    public isNonSupportedPlaceholder() {
+        if (checkPath(this.element, '["p:nvSpPr"][0]["p:nvPr"][0]["p:ph"][0]["$"]["type"]')) {
+            const type = getValueAtPath(
+                this.element,
+                '["p:nvSpPr"][0]["p:nvPr"][0]["p:ph"][0]["$"]["type"]'
+            );
+            if (!SUPPORTED_PLACEHOLDERS.includes(type)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public isPlaceholderListElement(slideLayoutTables): boolean {
+        if (checkPath(this.element, '["p:nvSpPr"][0]["p:nvPr"][0]["p:ph"][0]["$"]["idx"]')) {
+            const placeholderIdx = getValueAtPath(
+                this.element,
+                '["p:nvSpPr"][0]["p:nvPr"][0]["p:ph"][0]["$"]["idx"]'
+            );
+            let slideLayoutSpNode = undefined;
+            slideLayoutSpNode = slideLayoutTables["idxTable"][placeholderIdx];
+
+            return (
+                slideLayoutSpNode !== undefined &&
+                checkPath(slideLayoutSpNode, '["p:txBody"][0]["a:p"][0]["a:pPr"][0]["$"]["lvl"]') &&
+                checkPath(slideLayoutSpNode, '["p:txBody"][0]["a:lstStyle"][0]["a:lvl1pPr"][0]["a:buNone"]') === false
+            );
+        }
+
+        return false;
+    }
+
+    public getLayoutSpNodes(slideLayoutTables, slideMasterTables) {
+        const idx =
+            getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph"]) === undefined
+                ? undefined
+                : getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph", "$", "idx"]);
+        const type =
+            getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph"]) === undefined
+                ? undefined
+                : getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph", "$", "type"]);
+
+        let slideLayoutSpNode = undefined;
+        let slideMasterSpNode = undefined;
+
+        if (type !== undefined) {
+            slideLayoutSpNode = slideLayoutTables["typeTable"][type];
+            slideMasterSpNode = slideMasterTables["typeTable"][type];
+            return { slideLayoutSpNode, slideMasterSpNode };
```

Comment:
```suggestion

            return { slideLayoutSpNode, slideMasterSpNode };
```

Reply:
done

Reply:
but for some reason its not showing outdated, but i doubled check, i added the new line